crates/steel-core/src/rvals.rs

```diff
@@ -1073,52 +1074,52 @@ pub fn into_serializable_value(
 pub struct SteelMutableVector(pub(crate) Gc<RefCell<Vec<SteelVal>>>);
 
 #[derive(Clone, PartialEq, Eq)]
-pub struct SteelVector(pub(crate) Gc<im_rc::Vector<SteelVal>>);
+pub struct SteelVector(pub(crate) Gc<Vector<SteelVal>>);
 
 impl Deref for SteelVector {
-    type Target = im_rc::Vector<SteelVal>;
+    type Target = Vector<SteelVal>;
 
     fn deref(&self) -> &Self::Target {
         &self.0
     }
 }
 
-impl From<Gc<im_rc::Vector<SteelVal>>> for SteelVector {
-    fn from(value: Gc<im_rc::Vector<SteelVal>>) -> Self {
+impl From<Gc<Vector<SteelVal>>> for SteelVector {
+    fn from(value: Gc<Vector<SteelVal>>) -> Self {
         SteelVector(value)
     }
 }
 
 #[derive(Clone, PartialEq)]
-pub struct SteelHashMap(pub(crate) Gc<HashMap<SteelVal, SteelVal>>);
+pub struct SteelHashMap(pub(crate) Gc<ImmutableHashMap<SteelVal, SteelVal, FxBuildHasher>>);
 
 impl Deref for SteelHashMap {
-    type Target = HashMap<SteelVal, SteelVal>;
+    type Target = ImmutableHashMap<SteelVal, SteelVal, FxBuildHasher>;
 
     fn deref(&self) -> &Self::Target {
         &self.0
     }
 }
 
-impl From<Gc<HashMap<SteelVal, SteelVal>>> for SteelHashMap {
-    fn from(value: Gc<HashMap<SteelVal, SteelVal>>) -> Self {
+impl From<Gc<ImmutableHashMap<SteelVal, SteelVal, FxBuildHasher>>> for SteelHashMap {
+    fn from(value: Gc<ImmutableHashMap<SteelVal, SteelVal, FxBuildHasher>>) -> Self {
         SteelHashMap(value)
     }
 }
 
 #[derive(Clone, PartialEq)]
-pub struct SteelHashSet(pub(crate) Gc<im_rc::HashSet<SteelVal>>);
+pub struct SteelHashSet(pub(crate) Gc<im_rc::HashSet<SteelVal, FxBuildHasher>>);
```

Comment:
I realize this was unclear - however I would prefer if these do not take the `FxBuildHasher`